src/cpp/utils/SystemInfo.cpp

```diff
@@ -83,4 +86,76 @@ bool SystemInfo::file_exists(
     return (stat(filename.c_str(), &s) == 0 && s.st_mode & S_IFREG);
 }
 
+ReturnCode_t SystemInfo::load_environment_file(
+        const std::string& filename,
+        const std::string& env_name,
+        std::string& env_value)
+{
+    // Check that the file exists
+    if (!SystemInfo::file_exists(filename))
+    {
+        return ReturnCode_t::RETCODE_BAD_PARAMETER;
+    }
+
+    // Read json file
+    std::ifstream file(filename);
+    nlohmann::json file_content;
+
+    try
+    {
+        file >> file_content;
+    }
+    catch (const nlohmann::json::exception&)
+    {
+        return ReturnCode_t::RETCODE_ERROR;
+    }
+
+    try
+    {
+        env_value = file_content.at(env_name);
+        if (env_value.empty() || env_value.compare("") == 0)
+        {
+            return ReturnCode_t::RETCODE_NO_DATA;
+        }
+    }
+    catch (const nlohmann::json::exception&)
+    {
+        return ReturnCode_t::RETCODE_NO_DATA;
+    }
+    return ReturnCode_t::RETCODE_OK;
+}
+
+FileWatchHandle SystemInfo::watch_file(
+        std::string filename,
+        std::function<void(const std::string&)> callback)
+{
+#if defined(_WIN32) || defined(__unix__)
+    return FileWatchHandle (new filewatch::FileWatch<std::string>(filename,
+                   [callback](const std::string& path, const filewatch::Event change_type)
+                   {
+                       switch (change_type)
+                       {
+                           case filewatch::Event::modified:
+                               callback(path);
+                               break;
+                           default:
+                               // No-op
+                               break;
+                       }
+                   }));
+#endif // defined(_WIN32) || defined(__unix__)
+    (void)filename;
+    (void)callback;
```

Comment:
We are trying to use a more C++ style:

```suggestion
    static_cast<void>(filename);
    static_cast<void>(callback);
```

Reply:
Done in a2cf45ad9e5815e0f211b51ab1609c9eea6ee56a